src/FastSerialization/StreamReaderWriter.cs

```diff
@@ -647,8 +647,9 @@ public IOStreamStreamReader(string fileName, SerializationConfiguration config =
         /// The stream will be closed by the IOStreamStreamReader when it is closed.  
         /// </summary>
         public IOStreamStreamReader(Stream inputStream, int bufferSize = defaultBufferSize, bool leaveOpen = false, SerializationConfiguration config = null)
-            : base(new byte[bufferSize + align], 0, 0, config)
+            : base(new byte[bufferSize + (((int?)config?.StreamReaderAlignment) ?? 8)], 0, 0, config)
```

Comment:
I think I might be missing something here, but why does this need to be casted to int?.  I would expect this to just be an int, and not need the cast.

Reply:
`StreamReaderAlignment` is an enum where `EightBytes` is 8, etc.